src/mockServiceWorker.js

```diff
@@ -122,10 +122,13 @@ async function handleRequest(event, requestId) {
     ;(async function () {
       const responseClone = response.clone()
       // When performing original requests, response body will
-      // always be a ReadableStream, even for 204 responses.
+      // always be a ReadableStream, even for 204 / 304 responses.
       // But when creating a new Response instance on the client,
-      // the body for a 204 response must be null.
-      const responseBody = response.status === 204 ? null : responseClone.body
+      // the body for a 204 / 304 response must be null.
+      const responseBody =
+        response.status === 204 || response.status === 304
```

Comment:
When would a 304 response have body?

```js
new Response(null, { status: 304 }).body
// null
```

If you provide the body to the `Respnse`/`HttpResponse` constructor, it will throw in your response resolver directly (the standard behavior, not MSW). 

Can you tell me more about when this would happen that we need an explicit check? 

I think historically this was here to prevent a no-op responses declarations in response resolvers. But that's not possible now since we rely on the `Response` class directly, and it will throw when constructing an invalid response (if anything, perhaps we should remove this check altogether).

Reply:
A null body [status](https://fetch.spec.whatwg.org/#statuses) is a [status](https://fetch.spec.whatwg.org/#concept-status) that is 101, 103, 204, 205, or 304.

I think we probably want to check each of these statuses and ensure the response body is null for all of them?

Reply:
maybe worth something like

```ts
// outside of this scope
const nullBodyStatuses = new Set([101, 103, 204, 205, 304])

// instead of status === 204 ||| status === 304
nullBodyStatuses.has(response.status) ? null : responseJson.body
```

Reply:
> I think we probably want to check each of these statuses and ensure the response body is null for all of them?

Yes, with the correction that the Fetch API spec won't allow you to construct a 1xx response. No need to check against that. 

> maybe something like

We are getting this kind of logic repeated across MSW/Interceptors. Perhaps it's time we unified it and reused it across the libraries. 

This hardly qualifies as a standalone package though. We may expose this as a public API from the Interceptors since they are the common dependency here. Wdyt?

Reply:
I also realized this will be reproducible if the response has an empty string body. It will result in an empty readable stream, which the Response constructor still considers a present response body, thus throwing.

Reply:
> We are getting this kind of logic repeated across MSW/Interceptors. Perhaps it's time we unified it and reused it across the libraries.

Do we actually want to use that method here, since this is in the service worker, I don't believe we bundle imports directly (it has none currently)

Reply:
@mattcosta7, no, you're right, the worker would have to repeat that logic.